lib/hashie/mash.rb

```diff
@@ -74,8 +74,8 @@ def self.load(path, options = {})
       return @_mashes[path] if @_mashes.key?(path)
       raise ArgumentError, "The following file doesn't exist: #{path}" unless File.file?(path)
 
-      parser = options.fetch(:parser) { Hashie::Extensions::Parsers::YamlErbParser }
-      @_mashes[path] = new(parser.perform(path, options.except(:parser))).freeze
+      parser = options.delete(:parser) { Hashie::Extensions::Parsers::YamlErbParser }
```

Comment:
This modifies the calling hash, we'll need to at least `options.dup` first. I suggest injecting `except`, reimplementing `except` or using `reject`.

Reply:
This modifies the passed in options hash, which should just be a hash of Parser opts. This is a static method, so isn't called on an instance of any hash

Reply:
Yes, but future @dblock or @BobbyMcWho  will not know this and pass in an actual hash that will have a side effect that will be impossible to debug. We can't rely on users' good intentions.

Reply:
doesn't `parser.perform` ignore the `:parser` key? Maybe it's not necessary to modify the hash at all?

Reply:
Yeah, it doesn't look like this specific parser actually cares if that parser key is in there or not.